docs/proposals/dbaas/01-architecture/proposal.md

```diff
@@ -0,0 +1,695 @@
+     "kubernetes_cluster_name": "minikube"
+}'
+
+0.30s
+
+## Listing 20 database clusters
+
+time curl -k -s -o file1.out --request POST --url https://localhost:8443/v1/management/DBaaS/DBClusters/List --header 'accept: application/json' --header 'authorization: Basic YWRtaW46YWRtaW4=' --header 'content-type: application/json' --data '{
+     "kubernetes_cluster_name": "minikube"
+}'
+0.67s
+
+time curl -k -s -o file1.out --request POST --url https://localhost:8443/v1/management/DBaaS/DBClusters/List --header 'accept: application/json' --header 'authorization: Basic YWRtaW46YWRtaW4=' --header 'content-type: application/json' --data '{
+     "kubernetes_cluster_name": "minikube"
+}'
+1.14s
+
+time curl -k -s -o file1.out --request POST --url https://localhost:8443/v1/management/DBaaS/DBClusters/List --header 'accept: application/json' --header 'authorization: Basic YWRtaW46YWRtaW4=' --header 'content-type: application/json' --data '{
+     "kubernetes_cluster_name": "minikube"
+}'
+0.76s
+
+
+
+```
+The environment for the benchmark above uses the same testing environment
+
+1. PMM
+2. Remote kubernetes cluster
+3. Communicates either via dbaas-controller (first two test runs) or via dbaas-operator (the last test run)
+
+Note: Communication with dbaas-operator via client-go made without caches, however, using [client-go caches](https://pkg.go.dev/k8s.io/client-go/tools/cache) as well as [Limit and Continue](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#ListOptions) will improve performance for large scale deployments (more than 500 clusters created)
+
+
+### User Stories
+
+#### Story 1
+As an SRE person, I should be able to register the Kubernetes cluster using a service account without admin access to the cluster.
+
+#### Story 2
+As an SRE person, I should be able to understand what’s going wrong during the bootstrapping DBaaS feature inside of PMM in case of insufficient permissions so that I can debug and solve my issues. (E.g. No permissions to run kube-state-metrics, pxc, or psmdb operator).
+
+#### Story 3
+As an SRE person, I should be able to rename a Kubernetes cluster once it was provisioned automatically so that I can keep my naming conventions.
+
+#### Story 4
+As an SRE person, I should be able to specify which database operators I need to install in the cluster.
+
+#### Story 5
+As an SRE person, I should be able to create logical spaces to deploy databases so that I can easily split my environments. (e.g. dev namespace goes to the dev environment and the staging namespace goes to the staging environment. For the production environment I should be able to register and setup an additional cluster.)
+
+#### Story 6
+As an SRE person, I should be able to limit access to create/edit/destroy database clusters for specified users so that no devs are bugging me to do it for them.
```

Comment:
not sure how a proposed architecture meets this story.

Reply:
It can be implemented with k8s' RBAC integration